vhdbuilder/packer/buildperformance/pkg/service/helper.go

```diff
@@ -0,0 +1,198 @@
+	if err != nil {
+		return fmt.Errorf("error decoding local JSON file: %w", err)
+	}
+
+	key := "scripts"
+	raw := m[key]
+
+	holdingMap := map[string]map[string]string{}
+
+	err = json.Unmarshal(raw, &holdingMap)
+	if err != nil {
+		return fmt.Errorf("error unmarshalling local JSON file into temporary holding map")
+	}
+
+	err = maps.ConvertTimestampsToSeconds(holdingMap)
+	if err != nil {
+		return fmt.Errorf("failed to convert timestamps to floats for evaluation: %w", err)
+	}
+	return nil
+}
+
+func (maps *DataMaps) ConvertTimestampsToSeconds(holdingMap map[string]map[string]string) error {
+	for key, value := range holdingMap {
+		script := map[string]float64{}
+		for section, timeElapsed := range value {
+			t, err := time.Parse("15:04:05", timeElapsed)
+			if err != nil {
+				return fmt.Errorf("error parsing timestamp in local build JSON data: %w", err)
+			}
+			d := t.Sub(time.Date(t.Year(), t.Month(), t.Day(), 0, 0, 0, 0, t.Location()))
+			script[section] = d.Seconds()
+		}
+		maps.LocalPerformanceDataMap[key] = script
+	}
+	return nil
+}
+
+func (maps *DataMaps) ParseKustoData(data *SKU) error {
+	data.SKUPerformanceData = data.CleanData()
+	kustoData := []byte(data.SKUPerformanceData)
+	err := json.Unmarshal(kustoData, &maps.QueriedPerformanceDataMap)
+	if err != nil {
+		return fmt.Errorf("error unmarshalling kusto data: %w", err)
+	}
+	return nil
+}
+
+func (sku *SKU) CleanData() string {
+	var cleanedData string = strings.ReplaceAll(sku.SKUPerformanceData, "NaN", "-1")
+	return cleanedData
+}
+
+// After preparing performance data, evaluate it with EvaluatePerformance and associated helper functions
+func (maps *DataMaps) EvaluatePerformance() error {
+	// Iterate over LocalPerformanceDataMap and compare it against identical sections in QueriedPerformanceDataMap
+	for scriptName, scriptData := range maps.LocalPerformanceDataMap {
+		for section, timeElapsed := range scriptData {
+			// The value of QueriedPerformanceDataMap[scriptName][section] is an array with two elements: [avg, 2*stdev]
+			// First we check that the queried data contains this section
+			sectionDataSlice, ok := maps.QueriedPerformanceDataMap[scriptName][section]
+			if !ok {
+				fmt.Printf("No data available for %s in %s\n", section, scriptName)
+				continue
+			}
+			// Adding these together gives us the maximum time allowed for the section
+			maxTimeAllowed, err := SumSlice(sectionDataSlice)
+			if err != nil {
+				fmt.Printf("%v: %v", err, sectionDataSlice)
+				continue
+			}
+			if maxTimeAllowed == -1 {
+				fmt.Printf("Not enough data available for %s in %s\n", section, scriptName)
+				continue
+			}
+			if timeElapsed > maxTimeAllowed {
+				if maps.RegressionMap[scriptName] == nil {
+					maps.RegressionMap[scriptName] = map[string]float64{}
+				}
+				// Record the amount of time the section exceeded the maximum allowed time by
+				maps.RegressionMap[scriptName][section] = timeElapsed - maxTimeAllowed
+			}
+		}
+	}
```

Comment:
@jason1028kr good catch, thanks! Does the code change mitigate your concerns? On line 138 I added an access check on that key, and assuming it is successful, I then use the returned value for the rest of the section. 

The check for -1 returned from SumSlice ensures that enough data was available in Kusto to calculate both Stdev and mean. If either is not true, the CleanData function changes NaN values to -1 before this point so that the JSON can be unmarshalled and then evaluated in this function.

Reply:
Got it, thank you for the explanation. Yep, the change looks good!